### ISSUE TITLE
Pressing arrow up key at board causes panic at Asahi Ubuntu 24.04 at M1 Air

### ISSUE THREAD

**Issue by xet7** [yashs662/rust_kanban#20]
Hi,
when running at newest commit at https://github.com/yashs662/rust_kanban/commit/fffdd540d7714ff495dd19f5ce4e7c4293dd4527 , at M1 Air running Asahi Ubuntu 24.04 arm64, when at board I press arrow key up, it panics:

```
thread 'main' panicked at src/app/app_helper.rs:272:46:
attempt to subtract with overflow
stack backtrace:
   0:     0xaaaac56f587c - std::backtrace_rs::backtrace::libunwind::trace::h19f3f3b65d4800b3
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/../../backtrace/src/backtrace/libunwind.rs:116:5
   1:     0xaaaac56f587c - std::backtrace_rs::backtrace::trace_unsynchronized::hac80e876fd5e21d1
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0xaaaac56f587c - std::sys_common::backtrace::_print_fmt::hce9a5a3adb4e07fc
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:68:5
   3:     0xaaaac56f587c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h677fb81bc4f23286
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:44:22
   4:     0xaaaac571d43c - core::fmt::rt::Argument::fmt::h778d82891dc3b446
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/fmt/rt.rs:165:63
   5:     0xaaaac571d43c - core::fmt::write::h2b46371187d5b982
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/fmt/mod.rs:1168:21
   6:     0xaaaac56f1b50 - std::io::Write::write_fmt::h250845b38108f265
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/io/mod.rs:1835:15
   7:     0xaaaac56f56c4 - std::sys_common::backtrace::_print::h94309b499de845cb
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:47:5
   8:     0xaaaac56f56c4 - std::sys_common::backtrace::print::hf58378670760a884
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:34:9
   9:     0xaaaac56f6fbc - std::panicking::default_hook::{{closure}}::h60e1dc99a4ee0343
  10:     0xaaaac56f6c24 - std::panicking::default_hook::hc551fd63d402dcc8
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:298:9
  11:     0xaaaac4d32a78 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h30a8ce8cc26873bf
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/alloc/src/boxed.rs:2077:9
  12:     0xaaaac4d75850 - rust_kanban::main::{{closure}}::{{closure}}::haa2aa75489e94399
                               at /home/user/repos/rust_kanban/src/main.rs:49:13
  13:     0xaaaac56f74d8 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h20c0bb74d545fb12
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/alloc/src/boxed.rs:2077:9
  14:     0xaaaac56f74d8 - std::panicking::rust_panic_with_hook::h0985b2ea17fc89a2
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:799:13
  15:     0xaaaac56f7274 - std::panicking::begin_panic_handler::{{closure}}::h4c693bee5468b5eb
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:656:13
  16:     0xaaaac56f5d5c - std::sys_common::backtrace::__rust_end_short_backtrace::hea1747cffd850820
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:171:18
  17:     0xaaaac56f7014 - rust_begin_unwind
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:652:5
  18:     0xaaaac4ce5ef4 - core::panicking::panic_fmt::hd485fe77a335e582
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/panicking.rs:72:14
  19:     0xaaaac4ce6578 - core::panicking::panic_const::panic_const_sub_overflow::h540b26d7354a4e38
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/panicking.rs:179:21
  20:     0xaaaac4ec9df0 - rust_kanban::app::app_helper::go_up::hd3d3476fceb7619c
                               at /home/user/repos/rust_kanban/src/app/app_helper.rs:272:46
  21:     0xaaaac4d4b3e0 - rust_kanban::app::app_helper::handle_general_actions::{{closure}}::haddc9209f364a366
                               at /home/user/repos/rust_kanban/src/app/app_helper.rs:1303:29
  22:     0xaaaac4d56054 - rust_kanban::app::App::do_action::{{closure}}::h3cf3cce703c192ad
                               at /home/user/repos/rust_kanban/src/app/mod.rs:172:47
  23:     0xaaaac4d35fd4 - rust_kanban::util::start_ui::{{closure}}::h42aa99b59c712189
                               at /home/user/repos/rust_kanban/src/util.rs:63:66
  24:     0xaaaac4d750d8 - rust_kanban::main::{{closure}}::h4a0b5125e430c047
                               at /home/user/repos/rust_kanban/src/main.rs:116:32
  25:     0xaaaac4d78dbc - tokio::runtime::park::CachedParkThread::block_on::{{closure}}::h32af75b6214a6ec7
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/park.rs:281:63
  26:     0xaaaac4d788dc - tokio::runtime::coop::with_budget::h4465d0be4daec911
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/coop.rs:107:5
  27:     0xaaaac4d788dc - tokio::runtime::coop::budget::hca043d6e9fd87e9f
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/coop.rs:73:5
  28:     0xaaaac4d788dc - tokio::runtime::park::CachedParkThread::block_on::h943c758a19039d2d
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/park.rs:281:31
  29:     0xaaaac4d6dce8 - tokio::runtime::context::blocking::BlockingRegionGuard::block_on::h102b5a6dcd0dd71e
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/context/blocking.rs:66:9
  30:     0xaaaac4d5c340 - tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}::hbb78d2abd21ca18a
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/scheduler/multi_thread/mod.rs:87:13
  31:     0xaaaac4d6f970 - tokio::runtime::context::runtime::enter_runtime::haa70068330cca138
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/context/runtime.rs:65:16
  32:     0xaaaac4d5c2a4 - tokio::runtime::scheduler::multi_thread::MultiThread::block_on::hd3b1c0c3628e35e0
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/scheduler/multi_thread/mod.rs:86:9
  33:     0xaaaac4d5496c - tokio::runtime::runtime::Runtime::block_on_inner::h7f1bb619e8ec4365
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/runtime.rs:363:45
  34:     0xaaaac4d54a78 - tokio::runtime::runtime::Runtime::block_on::h2c9dceeabe7f84bd
                               at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.2/src/runtime/runtime.rs:335:13
  35:     0xaaaac4d2c8e4 - rust_kanban::main::h267037af4c7bc12f
                               at /home/user/repos/rust_kanban/src/main.rs:118:5
  36:     0xaaaac4ce7224 - core::ops::function::FnOnce::call_once::h5d365ca47f1458ea
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
  37:     0xaaaac4d6ded8 - std::sys_common::backtrace::__rust_begin_short_backtrace::hd373b0001654fcaa
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:155:18
  38:     0xaaaac4d75f84 - std::rt::lang_start::{{closure}}::ha0aa3714bfea6758
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:159:18
  39:     0xaaaac56e90e0 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h5b4a073b12e252f5
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:284:13
  40:     0xaaaac56e90e0 - std::panicking::try::do_call::h622f368d1d3adcbd
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  41:     0xaaaac56e90e0 - std::panicking::try::he0c2c4b5b9e4b307
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  42:     0xaaaac56e90e0 - std::panic::catch_unwind::hdddfa2979563acc4
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  43:     0xaaaac56e90e0 - std::rt::lang_start_internal::{{closure}}::h565136c6e381c89a
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:48
  44:     0xaaaac56e90e0 - std::panicking::try::do_call::h51eda94082ce84b4
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  45:     0xaaaac56e90e0 - std::panicking::try::heab4b6a452691063
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  46:     0xaaaac56e90e0 - std::panic::catch_unwind::hb144182404fb2ee5
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  47:     0xaaaac56e90e0 - std::rt::lang_start_internal::hcaaa9f13fcb6c389
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:20
  48:     0xaaaac4d75f54 - std::rt::lang_start::h2336cc8eac66b93d
                               at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:158:17
  49:     0xaaaac4d2c978 - main
  50:     0xffff959b84c4 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  51:     0xffff959b8598 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:360:3
  52:     0xaaaac4ce6830 - _start
  53:                0x0 - <unknown>
```

**Comment by yashs662** [comment]
Hmm that's odd I haven't touched that part of the code for months. Will look into it thanks for reporting

**Comment by yashs662** [comment]
Fixed in v0.10.4!!
